### PR TITLE
added new fidelis integration

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/14346

## Description
new integration for fidelis.
this is version 1.
not all commands are implemented due to problems in the API (they are marked as deprecated).

## Required version of Demisto
4.0.0

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [x] Documentation (https://github.com/demisto/etc/issues/15120)
- [ ] Code Review

## Additional changes
Describe additional changes done, for example adding a function to common server.

### CODE REVIEW

**Comment by anara123** [comment]
@yaakovi please fill the PR template according to our conventions

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 1]
```diff
@@ -0,0 +1,711 @@
...
+        elif command == 'fidelis-list-alerts':
+            fidelis_list_alerts_command()
+
+        elif command == 'fidelis-upload-pcap':
+            fidelis_upload_pcap_command()
+
+    except Exception, e:
+        LOG(e.message)
+        LOG.print_log()
+        raise
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: unreachable code

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+
+        elif command == 'fidelis-list-alerts':
+            fidelis_list_alerts_command()
+
+        elif command == 'fidelis-upload-pcap':
+            fidelis_upload_pcap_command()
+
+    except Exception, e:
+        LOG(e.message)
+        LOG.print_log()
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: we don't need this lines too

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 3]
```diff
@@ -0,0 +1,711 @@
...
+
+    if not demisto.params().get('proxy', False):
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+
+    ''' HELPER FUNCTIONS '''
+    def http_request(method, url_suffix, params=None, data=None):
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: params not used. either remove it or move it to be the last arg in the function

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 4]
```diff
@@ -0,0 +1,711 @@
...
+        if SESSION_ID is not None:
+            headers['x-uid'] = SESSION_ID
+            headers['Content-Type'] = 'application/json'
+
+        res = requests.request(
+            method,
+            SERVER_URL + url_suffix,
+            data=None if data is None else json.dumps(data),
+            headers=headers,
+            # params=params,
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: remove the comment if needed

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 5]
```diff
@@ -0,0 +1,777 @@
...
+  type: 9
+  required: true
+- display: Trust any certificate (unsecure)
+  name: unsecure
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: ""
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: defaultvalue true

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 6]
```diff
@@ -0,0 +1,777 @@
...
+  type: 0
+  required: true
+- display: Credentials
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Trust any certificate (unsecure)
+  name: unsecure
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: defaultvalue true

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 7]
```diff
@@ -0,0 +1,777 @@
...
+description: Automate Detection and Response to Network Threats and data leakage in
+  your organization with Fidelis Elevate Network Integration.
+configuration:
+- display: Server URL
+  name: server_url
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Credentials
+  name: credentials
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: check edge cases when url is incorrect, passowrd, username, proxy, insecure and print nice error

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 8]
```diff
@@ -0,0 +1,777 @@
...
+  required: false
+- display: Incident type
+  name: incidentType
+  defaultvalue: ""
+  type: 13
+  required: false
+- display: First fetch timestamp (<number> <time unit>, e.g., 12 hours, 7 days, 3     months,
+    1 year)
+  name: fetch_time
+  defaultvalue: 3 days
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: maybe we should consider to validate this param in test

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 9]
```diff
@@ -0,0 +1,777 @@
...
+
+
+    def fetch_incidents():
+        last_run = demisto.getLastRun()
+        # Get the last fetch time, if exists
+        last_fetch = last_run.get('time')
+
+        # Handle first time fetch, fetch incidents retroactively
+        if last_fetch is None:
+            last_fetch, _ = parse_date_range(FETCH_TIME, date_format='%Y-%m-%dT%H:%M:%SZ')
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: remove the Z

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 10]
```diff
@@ -0,0 +1,777 @@
...
+        latest = datetime.strptime(last_fetch, '%Y-%m-%dT%H:%M:%SZ')
+
+        incidents = []
+        items = fidelis_list_alerts(time_frame='Custom', start_time=last_fetch)
+        for item in items:
+            incident_date = datetime.strptime(item['ALERT_TIME'], '%Y-%m-%d %H:%M:%S')
+            incident = {
+                'Type': 'Fidelis',
+                'name': item['SUMMARY'],
+                'occurred': incident_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: remove the Z

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 11]
```diff
@@ -0,0 +1,777 @@
...
+                'rawJSON': json.dumps(item),
+            }
+            latest = max(latest, incident_date)
+            incidents.append(incident)
+
+        if latest != last_fetch:
+            last_fetch = (latest + timedelta(seconds=1)).strftime('%Y-%m-%dT%H:%M:%SZ')
+            demisto.setLastRun({'time' : last_fetch})
+
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: add a lot of demisto.debug() logs

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 12]
```diff
@@ -0,0 +1,777 @@
...
+        latest = datetime.strptime(last_fetch, '%Y-%m-%dT%H:%M:%SZ')
+
+        incidents = []
+        items = fidelis_list_alerts(time_frame='Custom', start_time=last_fetch)
+        for item in items:
+            incident_date = datetime.strptime(item['ALERT_TIME'], '%Y-%m-%d %H:%M:%S')
+            incident = {
+                'Type': 'Fidelis',
+                'name': item['SUMMARY'],
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: name could be item["Alert_ID"] + item["summary"]

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 13]
```diff
@@ -0,0 +1,777 @@
...
+      type: string
+    - contextPath: Fidelis.Alert.Type
+      description: Alert type
+      type: string
+    description: List open Alerts from Fidelis Elevate
+  - name: fidelis-upload-pcap
+    arguments:
+    - name: component_ip
+      required: true
+      description: Component IP
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: explain what is this and where I get these IPs

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 14]
```diff
@@ -0,0 +1,777 @@
+commonfields:
+  id: Fidelis Elevate Network
+  version: -1
+name: Fidelis Elevate Network
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: - is there an option to get the hash in malware alert
- separate upload-file to be upload-file and upload-url
-

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 15]
```diff
@@ -0,0 +1,777 @@
...
+        })
+
+
+    @logger
+    def fidelis_get_alert(alert_id):
+        return http_request('GET', '/j/rest/v1/alert/info/{}/'.format(alert_id))
+
+
+    def fidelis_delete_alert_command():
+        args = demisto.args()
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: image

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 16]
```diff
@@ -0,0 +1,777 @@
...
+    @logger
+    def fidelis_list_pcap_components():
+        res = http_request('GET', '/j/rest/policy/pcap/components/')
+
+        return res
+
+
+    def test_integration():
+        # the login is executed in the switch panel code
+        demisto.results('ok')
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: run get components

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 17]
```diff
@@ -0,0 +1,777 @@
...
+        data = {
+            'component': component_ip,
+            'files': file_names
+        }
+        res = http_request('POST', '/j/rest/policy/pcap/run/', data=data)
+        return 'Pcap file run submitted.'
+
+
+    def fidelis_list_pcap_components_command():
+        args = demisto.args()
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: remove

Comment (anara123) [reply Integrations/integration-FidelisElevateNetwork.yml]: no args

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 18]
```diff
@@ -0,0 +1,777 @@
...
+
+        results = fidelis_list_pcap_components()
+        ec = [{
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: from now on call ec => output

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 19]
```diff
@@ -0,0 +1,777 @@
...
+
+
+    @logger
+    def fidelis_run_pcap(component_ip, file_names):
+        data = {
+            'component': component_ip,
+            'files': file_names
+        }
+        res = http_request('POST', '/j/rest/policy/pcap/run/', data=data)
+        return 'Pcap file run submitted.'
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: dont return anything
move the message to _command

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 20]
```diff
@@ -0,0 +1,777 @@
...
+        # Handle error responses gracefully
+        if res.status_code not in {200, 201}:
+            if res.status_code == 500:
+                error = res.json()
+                return_error('Error in API call to Fidelis Integration [%d] - %s' % (res.status_code, error['detailMessage']))
+            else:
+                return_error('Error in API call to Fidelis Integration [%d] - %s' % (res.status_code, res.reason))
+
+        if is_json:
+            return res.json()
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: catch the error of the json parse.  If failed to parse json then print the error with the original res.content

File: Integrations/integration-FidelisElevateNetwork.yml [hunk 21]
```diff
@@ -0,0 +1,777 @@
...
+            headers=headers,
+            params=params,
+            files=files,
+            verify=not INSECURE,
+        )
+
+        # Handle error responses gracefully
+        if res.status_code not in {200, 201}:
+            if res.status_code == 500:
+                error = res.json()
```
Comment (anara123) [inline Integrations/integration-FidelisElevateNetwork.yml]: be careful here with the .json(), catch it also

File: Integrations/integration-FidelisElevateNetwork.yml (hunk 7 above)
Comment (Itay4) [reply Integrations/integration-FidelisElevateNetwork.yml]: please also check user with insufficient permissions case

**Pull request merged**